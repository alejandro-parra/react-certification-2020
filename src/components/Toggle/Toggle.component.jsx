import React, { useState } from 'react';
import Switch from '@material-ui/core/Switch';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';

const Toggle = () => {
    const [checked, setChecked] = useState(false);

    const toggleChecked = () => {
        setChecked((prev) => !prev);
    };

    return(
        <FormGroup>
        <FormControlLabel
          control={<Switch checked={checked} onChange={toggleChecked} />}
          label={checked ? 'Oscuro' : 'Claro'}
        />
      </FormGroup>
    );
}


export default Toggle;