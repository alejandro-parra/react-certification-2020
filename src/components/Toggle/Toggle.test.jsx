import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Toggle from './Toggle.component';
import '@testing-library/jest-dom/extend-expect';

describe('Toggle component', () => {
  it('renders properly', () => {
    const tree = renderer.create(<Toggle changeHandler={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('changes text after event and toggles', () => {
    const { getByText } = render(<Toggle changeHandler={() => {}} />);
    fireEvent.click(screen.getByRole('switch'));
    expect(getByText('Oscuro')).toBeTruthy();
  });
});
