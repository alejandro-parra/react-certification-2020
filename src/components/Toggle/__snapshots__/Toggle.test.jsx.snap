// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component renders properly 1`] = `
<div
  className="MuiFormGroup-root"
>
  <label
    className="MuiFormControlLabel-root"
    role="switch"
  >
    <span
      className="MuiSwitch-root"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiSwitch-switchBase MuiSwitch-colorSecondary"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <input
            className="PrivateSwitchBase-input-4 MuiSwitch-input"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb"
          />
        </span>
      </span>
      <span
        className="MuiSwitch-track"
      />
    </span>
    <span
      className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
    >
      Claro
    </span>
  </label>
</div>
`;
